tutorials/cloud-iot-firestore-config/functions/src/index.ts

```diff
@@ -13,51 +13,53 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 */
-
-
 'use strict';
 
 import cbor = require('cbor');
 
 import * as admin from "firebase-admin";
 import * as functions from 'firebase-functions';
-import { DeviceManager } from './devices';
-
-// create a device manager instance with a registry id, optionally pass a region
-const dm = new DeviceManager('config-demo');
+const iot = require('@google-cloud/iot');
+const client = new iot.v1.DeviceManagerClient();
```

Comment:
is the auth on this sync or async?

Reply:
What I had seen in other function use of libraries with async auth is that the first cold-start function would run before auth was set on the client.  Should be able to tell from any error logs on first cold start.

Reply:
From the log we can see that the initial auth is done sync before any request is fired to IoT Core:
cold-start run took 1355ms and subsequent call took 58ms. No error in the logs, which means no need of handling auth differently. 
` configUpdate 503653078556835 Function execution took 58 ms, finished with status: 'ok' D  configUpdate 503653078556835
 configUpdate 503653078556835 sample-device I  configUpdate 503653078556835
 configUpdate 503653078556835 Function execution started D  configUpdate 503653078556835
 configUpdate 503649745745940 Function execution took 1355 ms, finished with status: 'ok' D  configUpdate 503649745745940
 configUpdate 503649745745940 sample-device I  configUpdate 503649745745940
 configUpdate 503649745745940 Function execution started D  configUpdate 503649745745940
`